### ISSUE TITLE
Not optimal bike/cycling route

### ISSUE THREAD

**Issue by biodranik** [organicmaps/organicmaps#1201]
![S10827-131640](https://user-images.githubusercontent.com/170263/131214217-41fbd652-912b-42b9-9e2d-47dbab7c721b.jpg)
![S10827-13143019](https://user-images.githubusercontent.com/170263/131214218-d76139af-7ee8-4f72-b81d-4ffad2efab13.jpg)
![S10827-105418](https://user-images.githubusercontent.com/170263/131214220-9f52a3c4-ce40-4133-b466-eb513f4d7fa5.jpg)
![S10827-105448](https://user-images.githubusercontent.com/170263/131214225-0ed5e9d2-17b5-4f52-aadf-ba5c093b4e06.jpg)
![S10827-105455](https://user-images.githubusercontent.com/170263/131214226-0f87dca5-ede7-4e6d-9d6e-97858941f635.jpg)

**Comment by kikuok** [comment]
I confirm, this is the only big problem for me as well. It happens that walking and cycling routes are not built according to speed and I have to constantly help the program by setting intermediate points. And sometimes it even happens that the path leads across the lake! And the cycling route is always more suggested than walking, and therefore I almost always mark the walking route, although I move mostly by bicycle.

**Comment by biodranik** [comment]
Can you please share some examples for us to test and fix, ideally with a correct routing example?

**Comment by kikuok** [comment]
Yes, of course, but I think the problem is with the router that uses omaps. I try this route in app Locus maps and use offline navigation addiction - app Brouter - and this is find a shortest route instantly! Here are the coordinates of point a and point b if you need.(60.619347, 30.103713 and 
60.625555, 30.071129) but I will repeat that there are a lot of such examples on different parts of the map, and unfortunately it is mistake for more kilometers. The application shows the route by water, if the end point is not indicated on the map, that is, it is just a forest, not a path. I don't know if this is correct or not.
![Screenshot_02 09 21-19 10 23_Organic Maps](https://user-images.githubusercontent.com/81811653/131975788-f0a7adb6-c834-4062-b6d4-852d9fbaa9f2.png)
![Screenshot_02 09 21-19 10 34_Organic Maps](https://user-images.githubusercontent.com/81811653/131975821-09c36a7f-6396-4dcf-9e18-970893790f20.png)
![Screenshot_03 09 21-11 25 39_Locus Map](https://user-images.githubusercontent.com/81811653/131975852-25cb33a0-d54a-4307-b1e9-19344af099fa.png)

**Comment by vng** [comment]
One more feedback from user:

Another thing which recently annoyed me in Maps.me and which happens sadly in your app too, is the routing for bicycles. I am currently travelling by bike and many roads in the countryside of e.g. Texas are apparently defined as HWYs although they are overland country roads. Technically they are HWYs. What the routing does is trying to avoid those roads and to create a really stupid detour on dirt tracks through private land. Even trying to force the routing on the HWY with those three added points does not work for longer stretches.
The option of "avoid unpaved roads" in the routing settings does not change anything in the bicycle routing.
Do you have an idea how to fix that or can you try looking into that?
One example is the stretch between Marfa and Van Horn with this tiny village of Valentine in between. It's in Texas and you can try to get a route between either of those and you will see what I mean.

**Comment by vng** [comment]
Good idea here is to take into account "avoid_motorways" setting for bicycle (or pedestrian?) routing.

**Comment by biodranik** [comment]
I have two examples for you. Road from Marathon (30.205168, -103.244621) to Alpine (30.35738, -103.660938) in Texas. Please see the two Screenshots attached. The direct route is a two-lane countryside road defined as an HWY. The bicycle routing wants to get off that road whenever possible. It even uses probably just some trails to go through private dessert land. Trying to force it with the three stops works only partially.
The funny thing is, at another spot, where I cycled today it's opposite. It actually leads me on an Interstate four-lane HWY instead of following the rural country road. See Screenshot 3. Coordinates are 32.845995, -107.320545 to 32.953153, -107.311936.

![image](https://user-images.githubusercontent.com/170263/135903818-ced13ae4-4976-470e-a4bb-0e8ce46ce6a1.png)

![image](https://user-images.githubusercontent.com/170263/135903846-c85557e9-de88-4ab3-a66e-969bc675d8aa.png)

![image](https://user-images.githubusercontent.com/170263/135903864-9db668ce-d853-497b-837e-8ab7b2067dae.png)

**Comment by pastk** [comment]
Another interesting example (maybe worth its own issue?).

Build a bicycle route from Kaş `36.19939, 29.641353` to Ölüdeniz `36.571539, 29.142787`.
OM suggests to take 3(!) ferries crossing from Turkey to Greece and back!

![Screenshot_1637935595](https://user-images.githubusercontent.com/18434508/143594450-8fa9a251-b65d-4b88-919b-ec3de89c5fc0.png)

If I put an intermediate stop on the road:
![Screenshot_1637935740](https://user-images.githubusercontent.com/18434508/143594506-dffa2ed2-24d7-43ce-8f07-ce1891fb1afe.png)

There are several issues with that route:

1. probably border crossings are not taken into account, some options are:
* maybe there should be some default customs time allowance added like 1hr per border
* prefer route w/o border crossings by default;
* add "avoid border crossings" routing option

2. 90%+ of the route is by ferry while the user wants to travel by bicycle... so it's kind of not bicycle route anymore
* prefer routes with less ferry time if possible
* still, the 10h route that includes 1hr ferry should be preferred over 50h route without ferry - so needs some reasonable limits/weights set

**Comment by meinrad-0** [comment]
Encountering a similar problem as the last comment.
Routing either by bike or on foot, OM seems to prefer ferries over landroads. The examples are from Milazzo (38.249469, 15.247683) to Palermo (38.111215, 13.352446), Sicily, Italy. 

A solution would be to apply 'routings options' also for bike and hike.

![image](https://user-images.githubusercontent.com/94761015/148780021-9ba6010d-f9b0-4387-9c3a-e431e3adeb3b.png)

![image](https://user-images.githubusercontent.com/94761015/148780065-8da64e1c-657c-40e2-9ebd-99be680707b3.png)

**Comment by vng** [comment]
Many problems is that highway=footway/pedestrian are banned for cycling profile in many countries (see bicycle_model.cpp)
I don't know is it good or bad, but sometimes it makes really strange routes, e.g. bicycle in Istanbul via ferries is not working because of footways when entering/exit ferries.

**Comment by Turbotomate** [comment]
I found a severe issue with the bike-routing aswell:
The first screenshot is a way from Polterdamm 26, 38176 Bortfeld to Polterdamm 1, 38176 Bortfeld. The walking route is fine:
![Screenshot_20220405-111023_Organic Maps_1](https://user-images.githubusercontent.com/64365398/161721431-ba4b99b5-a596-40ae-b585-5d5e41ce1411.jpg)
Same start and destination, just switched to bike routing:
![Screenshot_20220405-111014_Organic Maps_1](https://user-images.githubusercontent.com/64365398/161721429-000abe94-e981-42d1-b53d-66dd9cd86dde.jpg)
As you can see the app takes a much longer way (3 times as long) which also has a worse condition for bikes. I checked OSM data and there is nothing that could prevent bikes from going the short way. When working with additional points on the route, you can get a route along the intended way.

Second example shows more problems in the same region. Start point is the same, destination more to the east. There is a perfect bike path, even a biking route in OSM data but the routing of organicmaps shows a way on a big road (100km/h) without a bike lane. You can see that it is much longer than the direct way aswell:
![Screenshot_20220405-111127_Organic Maps_1_1](https://user-images.githubusercontent.com/64365398/161721418-025e4001-70f4-4cf7-9424-ac0428a85d58.jpg)
This picture shows the same two points, routing for walking:
![Screenshot_20220405-111355_Organic Maps](https://user-images.githubusercontent.com/64365398/161721424-3db59070-3e51-48bb-abfc-44351bcafe63.jpg)
Here everything is calculated correctly. This would be the shorter, safer, better and intended way for biking aswell. Please find a fix, this are only two small examples but im having problems with bike routing all the time.

**Comment by vng** [comment]
@Turbotomate 
1. highway=footway; highway=pedestrian are not allowed for cycling in Germany by default.
See this OSM table https://wiki.openstreetmap.org/wiki/OSM_tags_for_routing/Access_restrictions#Germany

OM app is not so strict, but assigns 3 kmh speed (as dismount) for cycling on footways.
If you are sure that this footways are fully allowed for cycling, better to set bicycle=yes tag.
If you are not agree with defaults, probably (?) better to discuss it with local Germany OSM community.

2. Your issue is still valid, because bicycle routing can choose nearby highway=residential "Polterdamm" and "Glinderweg" road for cycling instead of highway=track. I will investigate it.

**Comment by Turbotomate** [comment]
You are totally correct, the footways there are not allowed for cycling. I expected the app to take the normal streets next to the footways like it should do. But your explanation explains to me, why it sometimes uses the street correctly and sometimes shows the long way around. -> When the part on the footway (3 km/h) becomes too long, the longer route around is faster. But like you said in 2., all of this shouldnt be an issue because there is the obvious bicycle-allowed street.

**Comment by vng** [comment]
@kikuok Your issue should be fixed with the next app update.

**Comment by vng** [comment]
- Texas looks ok, somebody put bicycle=yes on good primary road.
- Roswell, New Mexico bicycle routing by I 25 is valid because this motorway https://www.openstreetmap.org/way/38845962 has bicycle=yes tag. Looks suspicious, but anyway.

**Comment by vng** [comment]
@ray66 The app prefers "Hauptstraße" road [here](https://www.openstreetmap.org/?mlat=48.4162&mlon=9.8768#map=16/48.4162/9.8768&layers=Yl) because its also tagged as bicycle=yes and has surface=asphalt : https://www.openstreetmap.org/way/227562679

**Comment by vng** [comment]
@pastk Kaş -> Ölüdeniz works ok

**Comment by vng** [comment]
@meinrad-0 Your issue is valid and moved to the dedicated bicycle + ferry issue https://github.com/organicmaps/organicmaps/issues/2703
BTW, "Avoid ferry" routing option now should work with bicycles.

**Comment by keunes** [comment]
Hello,
I just had another similar case. Must say I didn't check OSM data so wouldn't know if the map data might be the cause.

From: om://44IerLqT2V/Shop
To: om://04Ieq1N2rr/Zwethkade_Zuid%2C_16

![Imagepipe_0.jpg](https://user-images.githubusercontent.com/11229646/178153882-1045c796-be50-4d1a-933e-177720abc329.jpg)

![Imagepipe_1.jpg](https://user-images.githubusercontent.com/11229646/178153888-a52c9dca-cd5c-43e6-b8f8-8eaaacbf6e24.jpg)

**Comment by Turbotomate** [comment]
> I found a severe issue with the bike-routing aswell: The first screenshot is a way from Polterdamm 26, 38176 Bortfeld to Polterdamm 1, 38176 Bortfeld. The walking route is fine: ![Screenshot_20220405-111023_Organic Maps_1](https://user-images.githubusercontent.com/64365398/161721431-ba4b99b5-a596-40ae-b585-5d5e41ce1411.jpg) Same start and destination, just switched to bike routing: ![Screenshot_20220405-111014_Organic Maps_1](https://user-images.githubusercontent.com/64365398/161721429-000abe94-e981-42d1-b53d-66dd9cd86dde.jpg) As you can see the app takes a much longer way (3 times as long) which also has a worse condition for bikes. I checked OSM data and there is nothing that could prevent bikes from going the short way. When working with additional points on the route, you can get a route along the intended way.
> 
> Second example shows more problems in the same region. Start point is the same, destination more to the east. There is a perfect bike path, even a biking route in OSM data but the routing of organicmaps shows a way on a big road (100km/h) without a bike lane. You can see that it is much longer than the direct way aswell: ![Screenshot_20220405-111127_Organic Maps_1_1](https://user-images.githubusercontent.com/64365398/161721418-025e4001-70f4-4cf7-9424-ac0428a85d58.jpg) This picture shows the same two points, routing for walking: ![Screenshot_20220405-111355_Organic Maps](https://user-images.githubusercontent.com/64365398/161721424-3db59070-3e51-48bb-abfc-44351bcafe63.jpg) Here everything is calculated correctly. This would be the shorter, safer, better and intended way for biking aswell. Please find a fix, this are only two small examples but im having problems with bike routing all the time.

My problems here finally got fixed with the new release. The app now shows the correct cycling path as i would have taken it. Thank you all!
Maybe some other people here should check their problems again, then we can close ths issue.

**Comment by CalebJohn** [comment]
It looks like streets with oneway=yes and oneway:bicycle=no are not always being routed correctly. If I adjust the start and/or end points I can force the router to make a route that passes through the street, but mostly it prefers to have me bike around the block. I double checked on [brouter](https://user-images.githubusercontent.com/2179547/233819656-c2edf0ec-5a84-44ae-aeff-7be88a4747a1.png) to confirm that it's not some map issue.
For testing I used:
start: https://osm.org/go/WJBxfoFx-?m=
end: https://osm.org/go/WJBxftTR-?m=

![bad_bike_route](https://user-images.githubusercontent.com/2179547/233819650-d7e4de72-ad40-4b4d-a7d5-3301335c8c9f.png)

I also have a screenshot showing how the router can be forced to follow [the expected route](https://user-images.githubusercontent.com/2179547/233819652-bd8513d8-07da-4fc7-8198-ce6f18a46695.png).

**Comment by biodranik** [comment]
@vng looks like for `oneway=yes` roads we're ignoring `oneway:bicycle=no` tags (from the example above). Is it easy to fix?

**Comment by map-per** [comment]
`oneway:bicycle=no` works in the places that I tested. I think It's the turn restriction (https://www.openstreetmap.org/relation/11977936) that's causing the problem mentioned above.

**Comment by pastk** [comment]
> I just had another similar case. Must say I didn't check OSM data so wouldn't know if the map data might be the cause.
> 
> From: om://44IerLqT2V/Shop To: om://04Ieq1N2rr/Zwethkade_Zuid%2C_16

Works fine in the current OM version
![image](https://github.com/user-attachments/assets/285ac88d-06b1-46df-829e-3a3f64491c60)

**Comment by pastk** [comment]
> Start position is Ulm, Germany, end position is Blaubeuren, Germany. The blue line I added to the screenshot shows the expected route. It follows paved cycleways and residential streets and is part of several bicycle route relations. Organic Maps ignores this route and goes by a [primary road](https://www.openstreetmap.org/way/157764819) and a [grade4 track](https://www.openstreetmap.org/way/26366597).
> 
> [Link to OSM Map](https://www.openstreetmap.org/?mlat=48.4162&mlon=9.8768#map=16/48.4162/9.8768&layers=Yl)

It was improved in https://github.com/organicmaps/organicmaps/pull/9692

![Image](https://github.com/user-attachments/assets/9c6d84ed-784f-404b-913d-30efed293354)
